include/oneapi/dpl/internal/async_impl/glue_async_impl.h

```diff
@@ -16,19 +16,23 @@
 #ifndef _ONEDPL_GLUE_ASYNC_IMPL_H
 #define _ONEDPL_GLUE_ASYNC_IMPL_H
 
-#include "../async_extension_defs.h"
+#ifdef _ONEDPL_BACKEND_SYCL
+#    include "../async_extension_defs.h"
 
-#if _ONEDPL_HETERO_BACKEND
-#    include "async_impl_hetero.h"
-#endif
+#    if _ONEDPL_HETERO_BACKEND
+#        include "async_impl_hetero.h"
+#    endif
+
+#    include "async_impl.h"
+#endif // _ONEDPL_BACKEND_SYCL
 
 namespace oneapi
 {
 namespace dpl
 {
 namespace experimental
 {
-
+#ifdef _ONEDPL_BACKEND_SYCL
```

Comment:
I think we should not to wrap the functions below in _ONEDPL_BACKEND_SYCL macro, because the functions are a kind of "general" - for host and hetero policies.
But I see that `__enable_if_device_execution_policy` is used here... 
Essentially, this file should be "universal" for host and device,, and should use just `__enable_if_execution_policy` like in `glue_algorithm_impl.h`
And should be `async_impl_hetero.h` for hetero async patterns and `async_impl.h` for host async patterns.

Reply:
Is it possible to do now, though? Ideally, yes, the files should be universal, but to the best of my knowledge we only have an implementation for device execution policies.

Reply:
As far as I understand, Sergey tries to address this issue:
> Compile error if we include <oneapi/dpl/async> and DONEDPL_BACKEND != dpcpp

The question is, what do we want to happen in this combination, which is essentially not supported? Perhaps the best thing is to add a `static_assert` with a meaningful error message.

Reply:
> Up to me in this case we should have ability to include headers in any order with any backend without errors.

I am not sure if it makes any sense, though, to include `<async>` if de-facto it cannot be used. Having such ability does not help code portability at all, in my opinion. 

For an example of that. look at `asynch.pass.cpp`. Yes you can include the header with no preprocessor conditions, but all the code in the test still has to be covered by the same conditions as before. I even think that having `#if...#endif` around the include makes it more clear that this functionality is not always available, and helps to understand the use of the same preprocessor checks later in the code.

In other words, I think the problem is not worth fixing at the moment, other than improving the diagnostics by adding a static assert directly into the header. Adding static asserts only when async functions are used seems to me too much (of work and extra code) for no practical value.

Reply:
Fixed:
- #if _ONEDPL_BACKEND_SYCL removed from
- - include/oneapi/dpl/internal/async_extension_defs.h
- - include/oneapi/dpl/internal/async_impl/glue_async_impl.h
- added check into include/oneapi/dpl/async :
```
#if !_ONEDPL_BACKEND_SYCL
#    error "Unable to include oneapi/dpl/async header because _ONEDPL_BACKEND_SYCL is not defined"
#endif // !_ONEDPL_BACKEND_SYCL
```
In tests now we include async header under define check:
```
#if TEST_DPCPP_BACKEND_PRESENT
#   include "oneapi/dpl/async"
#endif // TEST_DPCPP_BACKEND_PRESENT
```